File: java/Main2.java
public class Main2 {
    static int dangerous(int[] array, int index) {
        return array[index];
    }
    static int foo(int[] array, int counter) {
        if (counter == 0) return dangerous(array, counter + 9137);
        return foo(array, counter - 1);
    }
    public static void main(String[] args) {
        int[] a = new int[1000];
        int r = foo(a, 6);
        System.out.println("The result is " + r);
    }
}

File: kotlin/Main2.kt
object Main2 {
    @JvmStatic
    fun main(args: Array<String>) {
        val arr = IntArray(1000) { 0 }
        val r = foo(arr, 6)
        println("The result is $r")
    }

    fun dangerous(array: IntArray, index: Int): Int = array[index]

    fun foo(array: IntArray, counter: Int): Int =
        if (counter == 0) dangerous(array, counter + 9137)
        else foo(array, counter - 1)
}
